### PR TITLE
[docs]Secure Service-to-Service Communication

### PR DESCRIPTION
**Pull request by kaitlincarter-hc**
We created this new guide based on feedback from HashiConf Seattle. It walks users through service and proxy registration, creating intentions, and testing connections. It also provides some guidance on using Envoy and setting up Connect in production. 

Preview: https://deploy-preview-780--hashicorp-learn.netlify.com/consul/developer-mesh/connect-services

### CODE REVIEW

File: website/source/docs/guides/connect-services.html.md [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+  connect {
+    sidecar_service {}
+  }
+
+  check {
+    id       = "counting-check"
+    http     = "http://localhost:9003/health"
+    method   = "GET"
+    interval = "1s"
+    timeout  = "1s"
```
Comment (blake) [inline website/source/docs/guides/connect-services.html.md]: Only one of these values may be present on a check definition, not both. See https://github.com/hashicorp/consul/pull/6753.

File: website/source/docs/guides/connect-services.html.md [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+  }
+}
+```
+
+Services and sidecar proxies can be defined in either HCL or JSON. There is a
+[JSON version][counting-1.json] version of the service definition in the
```
Comment (blake) [inline website/source/docs/guides/connect-services.html.md]: The word "version" is repeated.

File: website/source/docs/guides/connect-services.html.md [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+        ]
+      }
+    }
+  }
+
+  check {
+    id       = "dashboard-check"
+    http     = "http://localhost:9002/health"
+    method   = "GET"
+    interval = "1s"
```
Comment (blake) [inline website/source/docs/guides/connect-services.html.md]: Same comment as on the counting service definition.

Comment (blake) [reply website/source/docs/guides/connect-services.html.md]: Disregard. My comment is incorrect.

File: website/source/docs/guides/connect-services.html.md (hunk 2 above)
Comment (kaitlincarter-hc) [reply website/source/docs/guides/connect-services.html.md]: updated

**Pull request merged**